src/services/s3.py

```diff
@@ -41,6 +41,8 @@ def check_if_bucket_accessible(self, bucket_name):
             return True
         except botocore.exceptions.ClientError:
             return False
+        except botocore.exceptions.EndpointConnectionError as err:
+            raise err
```

Comment:
what does this addition do? iiuc, this catches and immediately re-raises the error, so why include it?